<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>Introduction to omicwas</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Introduction to omicwas</h1>



<div id="installation" class="section level2">
<h2>Installation</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="kw">install.packages</span>(<span class="st">&quot;devtools&quot;</span>)</span>
<span id="cb1-2"><a href="#cb1-2"></a>devtools<span class="op">::</span><span class="kw">install_github</span>(<span class="st">&quot;fumi-github/omicwas&quot;</span>)</span></code></pre></div>
<p>If you encounter dependency error for <code>sva</code> package, install it from Bioconductor:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="cf">if</span> (<span class="op">!</span><span class="kw">requireNamespace</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="dt">quietly =</span> <span class="ot">TRUE</span>))</span>
<span id="cb2-2"><a href="#cb2-2"></a>   <span class="kw">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb2-3"><a href="#cb2-3"></a>BiocManager<span class="op">::</span><span class="kw">install</span>(<span class="st">&quot;sva&quot;</span>)</span></code></pre></div>
</div>
<div id="usage" class="section level2">
<h2>Usage</h2>
<p><code>omicwas</code> is a package for cell-type-specific disease association testing, using bulk tissue data as input. The package accepts DNA methylation data for epigenome-wide association studies (EWAS), as well as gene expression data for differential gene expression analyses. The main function is <code>ctassoc</code></p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a><span class="kw">library</span>(omicwas)</span></code></pre></div>
<p>See description.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a>?ctassoc</span></code></pre></div>
</div>
<div id="analyzing-dna-methylation" class="section level2">
<h2>Analyzing DNA methylation</h2>
<p>Let’s load a sample data.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a><span class="kw">data</span>(GSE42861small)</span>
<span id="cb5-2"><a href="#cb5-2"></a>X =<span class="st"> </span>GSE42861small<span class="op">$</span>X</span>
<span id="cb5-3"><a href="#cb5-3"></a>W =<span class="st"> </span>GSE42861small<span class="op">$</span>W</span>
<span id="cb5-4"><a href="#cb5-4"></a>Y =<span class="st"> </span>GSE42861small<span class="op">$</span>Y</span>
<span id="cb5-5"><a href="#cb5-5"></a>Y =<span class="st"> </span>Y[<span class="kw">seq</span>(<span class="dv">1</span>, <span class="dv">20</span>), ] <span class="co"># for brevity</span></span>
<span id="cb5-6"><a href="#cb5-6"></a>C =<span class="st"> </span>GSE42861small<span class="op">$</span>C</span></code></pre></div>
<p>See description.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>?GSE42861small</span></code></pre></div>
<p>The conventional way is to use ordinary linear regression.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a>result =<span class="st"> </span><span class="kw">ctassoc</span>(X, W, Y, <span class="dt">C =</span> C,</span>
<span id="cb7-2"><a href="#cb7-2"></a>                 <span class="dt">test =</span> <span class="st">&quot;full&quot;</span>)</span>
<span id="cb7-3"><a href="#cb7-3"></a><span class="co">#&gt; Linear regression ...</span></span>
<span id="cb7-4"><a href="#cb7-4"></a><span class="co">#&gt; Summarizing result ...</span></span>
<span id="cb7-5"><a href="#cb7-5"></a>result<span class="op">$</span>coefficients</span>
<span id="cb7-6"><a href="#cb7-6"></a><span class="co">#&gt; # A tibble: 380 x 6</span></span>
<span id="cb7-7"><a href="#cb7-7"></a><span class="co">#&gt;    response   celltype term    estimate statistic   p.value</span></span>
<span id="cb7-8"><a href="#cb7-8"></a><span class="co">#&gt;    &lt;chr&gt;      &lt;chr&gt;    &lt;chr&gt;      &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;</span></span>
<span id="cb7-9"><a href="#cb7-9"></a><span class="co">#&gt;  1 cg10543797 CD4.     disease   0.0116     0.387 6.99e-  1</span></span>
<span id="cb7-10"><a href="#cb7-10"></a><span class="co">#&gt;  2 cg10543797 CD4.     1         0.818     54.1   8.77e-241</span></span>
<span id="cb7-11"><a href="#cb7-11"></a><span class="co">#&gt;  3 cg10543797 CD8.     disease   0.0683     2.21  2.71e-  2</span></span>
<span id="cb7-12"><a href="#cb7-12"></a><span class="co">#&gt;  4 cg10543797 CD8.     1         0.798     51.0   1.45e-227</span></span>
<span id="cb7-13"><a href="#cb7-13"></a><span class="co">#&gt;  5 cg10543797 mono     disease  -0.0611    -0.977 3.29e-  1</span></span>
<span id="cb7-14"><a href="#cb7-14"></a><span class="co">#&gt;  6 cg10543797 mono     1         0.745     23.5   2.27e- 88</span></span>
<span id="cb7-15"><a href="#cb7-15"></a><span class="co">#&gt;  7 cg10543797 Bcells   disease   0.0469     0.761 4.47e-  1</span></span>
<span id="cb7-16"><a href="#cb7-16"></a><span class="co">#&gt;  8 cg10543797 Bcells   1         0.868     28.2   4.48e-114</span></span>
<span id="cb7-17"><a href="#cb7-17"></a><span class="co">#&gt;  9 cg10543797 NK       disease  -0.0272    -0.764 4.45e-  1</span></span>
<span id="cb7-18"><a href="#cb7-18"></a><span class="co">#&gt; 10 cg10543797 NK       1         0.817     43.6   7.22e-194</span></span>
<span id="cb7-19"><a href="#cb7-19"></a><span class="co">#&gt; # … with 370 more rows</span></span></code></pre></div>
<p>We recommend nonlinear regression with ridge regularization. For DNA methylation, we use the <strong>logit</strong> function for normalization, and the test option is <code>nls.logit</code></p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a>result =<span class="st"> </span><span class="kw">ctassoc</span>(X, W, Y, <span class="dt">C =</span> C,</span>
<span id="cb8-2"><a href="#cb8-2"></a>                 <span class="dt">test =</span> <span class="st">&quot;nls.logit&quot;</span>,</span>
<span id="cb8-3"><a href="#cb8-3"></a>                 <span class="dt">regularize =</span> <span class="ot">TRUE</span>)</span>
<span id="cb8-4"><a href="#cb8-4"></a><span class="co">#&gt; nls.logit ...</span></span>
<span id="cb8-5"><a href="#cb8-5"></a><span class="co">#&gt; </span></span>
<span id="cb8-6"><a href="#cb8-6"></a>  <span class="op">|</span><span class="st">                                                                            </span></span>
<span id="cb8-7"><a href="#cb8-7"></a><span class="st">  </span><span class="er">|</span><span class="st">                                                                      </span><span class="er">|</span><span class="st">   </span><span class="dv">0</span>%</span>
<span id="cb8-8"><a href="#cb8-8"></a>  <span class="op">|</span><span class="st">                                                                            </span></span>
<span id="cb8-9"><a href="#cb8-9"></a><span class="st">  </span><span class="er">|======================================================================|</span><span class="st"> </span><span class="dv">100</span>%</span>
<span id="cb8-10"><a href="#cb8-10"></a><span class="co">#&gt; Summarizing result ...</span></span>
<span id="cb8-11"><a href="#cb8-11"></a><span class="kw">print</span>(result<span class="op">$</span>coefficients, <span class="dt">n =</span> <span class="dv">20</span>)</span>
<span id="cb8-12"><a href="#cb8-12"></a><span class="co">#&gt; # A tibble: 380 x 6</span></span>
<span id="cb8-13"><a href="#cb8-13"></a><span class="co">#&gt;    response   celltype term                estimate statistic   p.value</span></span>
<span id="cb8-14"><a href="#cb8-14"></a><span class="co">#&gt;    &lt;chr&gt;      &lt;chr&gt;    &lt;chr&gt;                  &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;</span></span>
<span id="cb8-15"><a href="#cb8-15"></a><span class="co">#&gt;  1 cg10543797 CD4.     1                   1.50       15.5    1.24e- 46</span></span>
<span id="cb8-16"><a href="#cb8-16"></a><span class="co">#&gt;  2 cg10543797 CD8.     1                   1.33       14.5    9.53e- 42</span></span>
<span id="cb8-17"><a href="#cb8-17"></a><span class="co">#&gt;  3 cg10543797 mono     1                   1.19        6.93   1.03e- 11</span></span>
<span id="cb8-18"><a href="#cb8-18"></a><span class="co">#&gt;  4 cg10543797 Bcells   1                   1.81        7.29   9.12e- 13</span></span>
<span id="cb8-19"><a href="#cb8-19"></a><span class="co">#&gt;  5 cg10543797 NK       1                   1.63       13.5    8.63e- 37</span></span>
<span id="cb8-20"><a href="#cb8-20"></a><span class="co">#&gt;  6 cg10543797 Nue      1                   1.35       56.3    1.32e-253</span></span>
<span id="cb8-21"><a href="#cb8-21"></a><span class="co">#&gt;  7 cg10543797 Eos      1                   0.996       5.84   8.30e-  9</span></span>
<span id="cb8-22"><a href="#cb8-22"></a><span class="co">#&gt;  8 cg10543797 CD4.     disease             0.000711    0.465  6.42e-  1</span></span>
<span id="cb8-23"><a href="#cb8-23"></a><span class="co">#&gt;  9 cg10543797 CD8.     disease             0.00202     1.43   1.52e-  1</span></span>
<span id="cb8-24"><a href="#cb8-24"></a><span class="co">#&gt; 10 cg10543797 mono     disease            -0.000917   -0.921  3.57e-  1</span></span>
<span id="cb8-25"><a href="#cb8-25"></a><span class="co">#&gt; 11 cg10543797 Bcells   disease             0.000350    0.505  6.14e-  1</span></span>
<span id="cb8-26"><a href="#cb8-26"></a><span class="co">#&gt; 12 cg10543797 NK       disease            -0.000504   -0.468  6.40e-  1</span></span>
<span id="cb8-27"><a href="#cb8-27"></a><span class="co">#&gt; 13 cg10543797 Nue      disease            -0.00555    -0.855  3.93e-  1</span></span>
<span id="cb8-28"><a href="#cb8-28"></a><span class="co">#&gt; 14 cg10543797 Eos      disease             0.000278    0.474  6.35e-  1</span></span>
<span id="cb8-29"><a href="#cb8-29"></a><span class="co">#&gt; 15 cg10543797 &lt;NA&gt;     sexF               -0.0415     -3.94   9.16e-  5</span></span>
<span id="cb8-30"><a href="#cb8-30"></a><span class="co">#&gt; 16 cg10543797 &lt;NA&gt;     age                -0.000303   -0.746  4.56e-  1</span></span>
<span id="cb8-31"><a href="#cb8-31"></a><span class="co">#&gt; 17 cg10543797 &lt;NA&gt;     smoking_current    -0.00532    -0.430  6.67e-  1</span></span>
<span id="cb8-32"><a href="#cb8-32"></a><span class="co">#&gt; 18 cg10543797 &lt;NA&gt;     smoking_ex         -0.0192     -1.58   1.13e-  1</span></span>
<span id="cb8-33"><a href="#cb8-33"></a><span class="co">#&gt; 19 cg10543797 &lt;NA&gt;     smoking_occasional  0.00109     0.0627 9.50e-  1</span></span>
<span id="cb8-34"><a href="#cb8-34"></a><span class="co">#&gt; 20 cg22718169 CD4.     1                   2.18       15.5    2.75e- 46</span></span>
<span id="cb8-35"><a href="#cb8-35"></a><span class="co">#&gt; # … with 360 more rows</span></span></code></pre></div>
<p>The first 19 lines show the result for CpG site cg10543797. Line 1 shows that the basal methylation level in CD4. (actually CD4+ T cells) is plogis(1.498) = 0.817, so this cell type is 81% methylated. Line 8 shows that the CD4.-specific effect of the disease is 7.10e-04 (in logit scale). Since the p.value is 0.64, this is not significant. Line 15 shows that the effect of sexF (female compared to male) is -4.14e-02 with a small p.value 9.15e-05. Since sexF is a covariate that has uniform effect across cell types, the celltype column is NA.</p>
</div>
<div id="analyzing-gene-expression" class="section level2">
<h2>Analyzing gene expression</h2>
<p>Let’s load a sample data.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a><span class="kw">data</span>(GTExsmall)</span>
<span id="cb9-2"><a href="#cb9-2"></a>X =<span class="st"> </span>GTExsmall<span class="op">$</span>X</span>
<span id="cb9-3"><a href="#cb9-3"></a>W =<span class="st"> </span>GTExsmall<span class="op">$</span>W</span>
<span id="cb9-4"><a href="#cb9-4"></a>Y =<span class="st"> </span>GTExsmall<span class="op">$</span>Y <span class="op">+</span><span class="st"> </span><span class="dv">1</span></span>
<span id="cb9-5"><a href="#cb9-5"></a>Y =<span class="st"> </span>Y[<span class="kw">seq</span>(<span class="dv">1</span>, <span class="dv">20</span>), ] <span class="co"># for brevity</span></span>
<span id="cb9-6"><a href="#cb9-6"></a>C =<span class="st"> </span>GTExsmall<span class="op">$</span>C</span></code></pre></div>
<p>See description.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a>?GTExsmall</span></code></pre></div>
<p>The conventional way is to use ordinary linear regression.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a>result =<span class="st"> </span><span class="kw">ctassoc</span>(X, W, Y, <span class="dt">C =</span> C,</span>
<span id="cb11-2"><a href="#cb11-2"></a>                 <span class="dt">test =</span> <span class="st">&quot;full&quot;</span>)</span>
<span id="cb11-3"><a href="#cb11-3"></a><span class="co">#&gt; Linear regression ...</span></span>
<span id="cb11-4"><a href="#cb11-4"></a><span class="co">#&gt; Summarizing result ...</span></span>
<span id="cb11-5"><a href="#cb11-5"></a>result<span class="op">$</span>coefficients</span>
<span id="cb11-6"><a href="#cb11-6"></a><span class="co">#&gt; # A tibble: 260 x 6</span></span>
<span id="cb11-7"><a href="#cb11-7"></a><span class="co">#&gt;    response        celltype              term   estimate statistic  p.value</span></span>
<span id="cb11-8"><a href="#cb11-8"></a><span class="co">#&gt;    &lt;chr&gt;           &lt;chr&gt;                 &lt;chr&gt;     &lt;dbl&gt;     &lt;dbl&gt;    &lt;dbl&gt;</span></span>
<span id="cb11-9"><a href="#cb11-9"></a><span class="co">#&gt;  1 ENSG00000059804 Granulocytes          age      45611.     2.56  1.07e- 2</span></span>
<span id="cb11-10"><a href="#cb11-10"></a><span class="co">#&gt;  2 ENSG00000059804 Granulocytes          1       378100.     1.64  1.02e- 1</span></span>
<span id="cb11-11"><a href="#cb11-11"></a><span class="co">#&gt;  3 ENSG00000059804 B.cells..CD19..       age     143773.     0.683 4.95e- 1</span></span>
<span id="cb11-12"><a href="#cb11-12"></a><span class="co">#&gt;  4 ENSG00000059804 B.cells..CD19..       1      -774328.    -0.324 7.46e- 1</span></span>
<span id="cb11-13"><a href="#cb11-13"></a><span class="co">#&gt;  5 ENSG00000059804 CD4..T.cells          age    -104302.    -2.42  1.59e- 2</span></span>
<span id="cb11-14"><a href="#cb11-14"></a><span class="co">#&gt;  6 ENSG00000059804 CD4..T.cells          1     -1043816.    -2.28  2.30e- 2</span></span>
<span id="cb11-15"><a href="#cb11-15"></a><span class="co">#&gt;  7 ENSG00000059804 CD8..T.cells          age      94351.     1.42  1.57e- 1</span></span>
<span id="cb11-16"><a href="#cb11-16"></a><span class="co">#&gt;  8 ENSG00000059804 CD8..T.cells          1      2577296.     3.49  5.47e- 4</span></span>
<span id="cb11-17"><a href="#cb11-17"></a><span class="co">#&gt;  9 ENSG00000059804 NK.cells..CD3..CD56.. age     134268.     2.34  1.96e- 2</span></span>
<span id="cb11-18"><a href="#cb11-18"></a><span class="co">#&gt; 10 ENSG00000059804 NK.cells..CD3..CD56.. 1      5027033.     6.61  1.33e-10</span></span>
<span id="cb11-19"><a href="#cb11-19"></a><span class="co">#&gt; # … with 250 more rows</span></span></code></pre></div>
<p>We recommend nonlinear regression with ridge regularization. For DNA methylation, we use the <strong>log</strong> function for normalization, and the test option is <code>nls.log</code></p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a>result =<span class="st"> </span><span class="kw">ctassoc</span>(X, W, Y, <span class="dt">C =</span> C,</span>
<span id="cb12-2"><a href="#cb12-2"></a>                 <span class="dt">test =</span> <span class="st">&quot;nls.log&quot;</span>,</span>
<span id="cb12-3"><a href="#cb12-3"></a>                 <span class="dt">regularize =</span> <span class="ot">TRUE</span>)</span>
<span id="cb12-4"><a href="#cb12-4"></a><span class="co">#&gt; nls.log ...</span></span>
<span id="cb12-5"><a href="#cb12-5"></a><span class="co">#&gt; </span></span>
<span id="cb12-6"><a href="#cb12-6"></a>  <span class="op">|</span><span class="st">                                                                            </span></span>
<span id="cb12-7"><a href="#cb12-7"></a><span class="st">  </span><span class="er">|</span><span class="st">                                                                      </span><span class="er">|</span><span class="st">   </span><span class="dv">0</span>%</span>
<span id="cb12-8"><a href="#cb12-8"></a>  <span class="op">|</span><span class="st">                                                                            </span></span>
<span id="cb12-9"><a href="#cb12-9"></a><span class="st">  </span><span class="er">|======================================================================|</span><span class="st"> </span><span class="dv">100</span>%</span>
<span id="cb12-10"><a href="#cb12-10"></a><span class="co">#&gt; Summarizing result ...</span></span>
<span id="cb12-11"><a href="#cb12-11"></a><span class="kw">print</span>(result<span class="op">$</span>coefficients, <span class="dt">n =</span> <span class="dv">15</span>)</span>
<span id="cb12-12"><a href="#cb12-12"></a><span class="co">#&gt; # A tibble: 260 x 6</span></span>
<span id="cb12-13"><a href="#cb12-13"></a><span class="co">#&gt;    response        celltype              term    estimate statistic   p.value</span></span>
<span id="cb12-14"><a href="#cb12-14"></a><span class="co">#&gt;    &lt;chr&gt;           &lt;chr&gt;                 &lt;chr&gt;      &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;</span></span>
<span id="cb12-15"><a href="#cb12-15"></a><span class="co">#&gt;  1 ENSG00000059804 Granulocytes          1      8.85         0.544  5.87e-  1</span></span>
<span id="cb12-16"><a href="#cb12-16"></a><span class="co">#&gt;  2 ENSG00000059804 B.cells..CD19..       1      8.85         0.0490 9.61e-  1</span></span>
<span id="cb12-17"><a href="#cb12-17"></a><span class="co">#&gt;  3 ENSG00000059804 CD4..T.cells          1      8.85         0.252  8.01e-  1</span></span>
<span id="cb12-18"><a href="#cb12-18"></a><span class="co">#&gt;  4 ENSG00000059804 CD8..T.cells          1     10.7          1.21   2.25e-  1</span></span>
<span id="cb12-19"><a href="#cb12-19"></a><span class="co">#&gt;  5 ENSG00000059804 NK.cells..CD3..CD56.. 1     14.2         52.7    2.61e-179</span></span>
<span id="cb12-20"><a href="#cb12-20"></a><span class="co">#&gt;  6 ENSG00000059804 Monocytes..CD14..     1      8.85         0.0666 9.47e-  1</span></span>
<span id="cb12-21"><a href="#cb12-21"></a><span class="co">#&gt;  7 ENSG00000059804 Granulocytes          age    0.000462     0.221  8.25e-  1</span></span>
<span id="cb12-22"><a href="#cb12-22"></a><span class="co">#&gt;  8 ENSG00000059804 B.cells..CD19..       age    0.0000111    0.107  9.15e-  1</span></span>
<span id="cb12-23"><a href="#cb12-23"></a><span class="co">#&gt;  9 ENSG00000059804 CD4..T.cells          age   -0.0000308   -0.0401 9.68e-  1</span></span>
<span id="cb12-24"><a href="#cb12-24"></a><span class="co">#&gt; 10 ENSG00000059804 CD8..T.cells          age    0.000655     0.312  7.55e-  1</span></span>
<span id="cb12-25"><a href="#cb12-25"></a><span class="co">#&gt; 11 ENSG00000059804 NK.cells..CD3..CD56.. age    0.0185       4.62   5.20e-  6</span></span>
<span id="cb12-26"><a href="#cb12-26"></a><span class="co">#&gt; 12 ENSG00000059804 Monocytes..CD14..     age    0.0000263    0.0955 9.24e-  1</span></span>
<span id="cb12-27"><a href="#cb12-27"></a><span class="co">#&gt; 13 ENSG00000059804 &lt;NA&gt;                  sexF  -0.00257     -0.0273 9.78e-  1</span></span>
<span id="cb12-28"><a href="#cb12-28"></a><span class="co">#&gt; 14 ENSG00000147454 Granulocytes          1     10.1          5.20   3.20e-  7</span></span>
<span id="cb12-29"><a href="#cb12-29"></a><span class="co">#&gt; 15 ENSG00000147454 B.cells..CD19..       1      9.25         0.197  8.44e-  1</span></span>
<span id="cb12-30"><a href="#cb12-30"></a><span class="co">#&gt; # … with 245 more rows</span></span></code></pre></div>
<p>The first 13 lines show the result for transcript ENSG00000059804. Line 1 shows that the basal expression level in Granulocytes is exp(8.847) = 6955. Line 7 shows that the Granulocytes-specific effect of age is 4.62e-04 (in log scale). Since the p.value is 0.82, this is not significant. Line 13 shows that the effect of sexF (female compared to male) is -2.57e-03 with p.value 0.97 Since sexF is a covariate that has uniform effect across cell types, the celltype column is NA.</p>
</div>
<div id="analyzing-mqtl" class="section level2">
<h2>Analyzing mQTL</h2>
<p>For QTL analyses, we use <code>ctcisQTL</code> function instead of <code>ctassoc</code>. To speed up computation, we perform linear ridge regression, thus the statistical test is almost identical to <code>ctassoc(test = &quot;nls.identity&quot;, regularize = TRUE)</code>. We analyze only in the linear scale. Association analysis is performed between each row of Y and each row of X. See description.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a>?ctcisQTL</span></code></pre></div>
<p>Let’s load a sample data.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a><span class="kw">data</span>(GSE79262small)</span>
<span id="cb14-2"><a href="#cb14-2"></a>X    =<span class="st"> </span>GSE79262small<span class="op">$</span>X</span>
<span id="cb14-3"><a href="#cb14-3"></a>Xpos =<span class="st"> </span>GSE79262small<span class="op">$</span>Xpos</span>
<span id="cb14-4"><a href="#cb14-4"></a>W    =<span class="st"> </span>GSE79262small<span class="op">$</span>W</span>
<span id="cb14-5"><a href="#cb14-5"></a>Y    =<span class="st"> </span>GSE79262small<span class="op">$</span>Y</span>
<span id="cb14-6"><a href="#cb14-6"></a>Ypos =<span class="st"> </span>GSE79262small<span class="op">$</span>Ypos</span>
<span id="cb14-7"><a href="#cb14-7"></a>C    =<span class="st"> </span>GSE79262small<span class="op">$</span>C</span>
<span id="cb14-8"><a href="#cb14-8"></a>X    =<span class="st"> </span>X[<span class="kw">seq</span>(<span class="dv">1</span>, <span class="dv">3001</span>, <span class="dv">100</span>), ] <span class="co"># for brevity</span></span>
<span id="cb14-9"><a href="#cb14-9"></a>Xpos =<span class="st"> </span>Xpos[<span class="kw">seq</span>(<span class="dv">1</span>, <span class="dv">3001</span>, <span class="dv">100</span>)]</span>
<span id="cb14-10"><a href="#cb14-10"></a>Y    =<span class="st"> </span>Y[<span class="kw">seq</span>(<span class="dv">1</span>, <span class="dv">501</span>, <span class="dv">100</span>), ]</span>
<span id="cb14-11"><a href="#cb14-11"></a>Ypos =<span class="st"> </span>Ypos[<span class="kw">seq</span>(<span class="dv">1</span>, <span class="dv">501</span>, <span class="dv">100</span>)]</span></code></pre></div>
<p>See description.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a>?GSE79262small</span></code></pre></div>
<p>Analyze mQTL.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a><span class="kw">ctcisQTL</span>(X, Xpos, W, Y, Ypos, <span class="dt">C =</span> C)</span>
<span id="cb16-2"><a href="#cb16-2"></a><span class="co">#&gt; Writing output to /var/folders/7q/n5cckft14d9345z_tdpy3qqc0000gn/T//Rtmpg55IpF/ctcisQTL.out.txt</span></span>
<span id="cb16-3"><a href="#cb16-3"></a><span class="co">#&gt; </span></span>
<span id="cb16-4"><a href="#cb16-4"></a>  <span class="op">|</span><span class="st">                                                                            </span></span>
<span id="cb16-5"><a href="#cb16-5"></a><span class="st">  </span><span class="er">|</span><span class="st">                                                                      </span><span class="er">|</span><span class="st">   </span><span class="dv">0</span>%</span>
<span id="cb16-6"><a href="#cb16-6"></a>  <span class="op">|</span><span class="st">                                                                            </span></span>
<span id="cb16-7"><a href="#cb16-7"></a><span class="st">  </span><span class="er">|==</span><span class="st">                                                                    </span><span class="er">|</span><span class="st">   </span><span class="dv">3</span>%</span>
<span id="cb16-8"><a href="#cb16-8"></a>  <span class="op">|</span><span class="st">                                                                            </span></span>
<span id="cb16-9"><a href="#cb16-9"></a><span class="st">  </span><span class="er">|=====</span><span class="st">                                                                 </span><span class="er">|</span><span class="st">   </span><span class="dv">6</span>%</span>
<span id="cb16-10"><a href="#cb16-10"></a>  <span class="op">|</span><span class="st">                                                                            </span></span>
<span id="cb16-11"><a href="#cb16-11"></a><span class="st">  </span><span class="er">|=======</span><span class="st">                                                               </span><span class="er">|</span><span class="st">  </span><span class="dv">10</span>%</span>
<span id="cb16-12"><a href="#cb16-12"></a>  <span class="op">|</span><span class="st">                                                                            </span></span>
<span id="cb16-13"><a href="#cb16-13"></a><span class="st">  </span><span class="er">|=========</span><span class="st">                                                             </span><span class="er">|</span><span class="st">  </span><span class="dv">13</span>%</span>
<span id="cb16-14"><a href="#cb16-14"></a>  <span class="op">|</span><span class="st">                                                                            </span></span>
<span id="cb16-15"><a href="#cb16-15"></a><span class="st">  </span><span class="er">|===========</span><span class="st">                                                           </span><span class="er">|</span><span class="st">  </span><span class="dv">16</span>%</span>
<span id="cb16-16"><a href="#cb16-16"></a>  <span class="op">|</span><span class="st">                                                                            </span></span>
<span id="cb16-17"><a href="#cb16-17"></a><span class="st">  </span><span class="er">|==============</span><span class="st">                                                        </span><span class="er">|</span><span class="st">  </span><span class="dv">19</span>%</span>
<span id="cb16-18"><a href="#cb16-18"></a>  <span class="op">|</span><span class="st">                                                                            </span></span>
<span id="cb16-19"><a href="#cb16-19"></a><span class="st">  </span><span class="er">|================</span><span class="st">                                                      </span><span class="er">|</span><span class="st">  </span><span class="dv">23</span>%</span>
<span id="cb16-20"><a href="#cb16-20"></a>  <span class="op">|</span><span class="st">                                                                            </span></span>
<span id="cb16-21"><a href="#cb16-21"></a><span class="st">  </span><span class="er">|==================</span><span class="st">                                                    </span><span class="er">|</span><span class="st">  </span><span class="dv">26</span>%</span>
<span id="cb16-22"><a href="#cb16-22"></a>  <span class="op">|</span><span class="st">                                                                            </span></span>
<span id="cb16-23"><a href="#cb16-23"></a><span class="st">  </span><span class="er">|====================</span><span class="st">                                                  </span><span class="er">|</span><span class="st">  </span><span class="dv">29</span>%</span>
<span id="cb16-24"><a href="#cb16-24"></a>  <span class="op">|</span><span class="st">                                                                            </span></span>
<span id="cb16-25"><a href="#cb16-25"></a><span class="st">  </span><span class="er">|=======================</span><span class="st">                                               </span><span class="er">|</span><span class="st">  </span><span class="dv">32</span>%</span>
<span id="cb16-26"><a href="#cb16-26"></a>  <span class="op">|</span><span class="st">                                                                            </span></span>
<span id="cb16-27"><a href="#cb16-27"></a><span class="st">  </span><span class="er">|================================</span><span class="st">                                      </span><span class="er">|</span><span class="st">  </span><span class="dv">45</span>%</span>
<span id="cb16-28"><a href="#cb16-28"></a>  <span class="op">|</span><span class="st">                                                                            </span></span>
<span id="cb16-29"><a href="#cb16-29"></a><span class="st">  </span><span class="er">|==================================</span><span class="st">                                    </span><span class="er">|</span><span class="st">  </span><span class="dv">48</span>%</span>
<span id="cb16-30"><a href="#cb16-30"></a>  <span class="op">|</span><span class="st">                                                                            </span></span>
<span id="cb16-31"><a href="#cb16-31"></a><span class="st">  </span><span class="er">|====================================</span><span class="st">                                  </span><span class="er">|</span><span class="st">  </span><span class="dv">52</span>%</span>
<span id="cb16-32"><a href="#cb16-32"></a>  <span class="op">|</span><span class="st">                                                                            </span></span>
<span id="cb16-33"><a href="#cb16-33"></a><span class="st">  </span><span class="er">|======================================</span><span class="st">                                </span><span class="er">|</span><span class="st">  </span><span class="dv">55</span>%</span>
<span id="cb16-34"><a href="#cb16-34"></a>  <span class="op">|</span><span class="st">                                                                            </span></span>
<span id="cb16-35"><a href="#cb16-35"></a><span class="st">  </span><span class="er">|=========================================</span><span class="st">                             </span><span class="er">|</span><span class="st">  </span><span class="dv">58</span>%</span>
<span id="cb16-36"><a href="#cb16-36"></a>  <span class="op">|</span><span class="st">                                                                            </span></span>
<span id="cb16-37"><a href="#cb16-37"></a><span class="st">  </span><span class="er">|===========================================================</span><span class="st">           </span><span class="er">|</span><span class="st">  </span><span class="dv">84</span>%</span>
<span id="cb16-38"><a href="#cb16-38"></a>  <span class="op">|</span><span class="st">                                                                            </span></span>
<span id="cb16-39"><a href="#cb16-39"></a><span class="st">  </span><span class="er">|=============================================================</span><span class="st">         </span><span class="er">|</span><span class="st">  </span><span class="dv">87</span>%</span>
<span id="cb16-40"><a href="#cb16-40"></a>  <span class="op">|</span><span class="st">                                                                            </span></span>
<span id="cb16-41"><a href="#cb16-41"></a><span class="st">  </span><span class="er">|===============================================================</span><span class="st">       </span><span class="er">|</span><span class="st">  </span><span class="dv">90</span>%</span>
<span id="cb16-42"><a href="#cb16-42"></a>  <span class="op">|</span><span class="st">                                                                            </span></span>
<span id="cb16-43"><a href="#cb16-43"></a><span class="st">  </span><span class="er">|=================================================================</span><span class="st">     </span><span class="er">|</span><span class="st">  </span><span class="dv">94</span>%</span>
<span id="cb16-44"><a href="#cb16-44"></a>  <span class="op">|</span><span class="st">                                                                            </span></span>
<span id="cb16-45"><a href="#cb16-45"></a><span class="st">  </span><span class="er">|====================================================================</span><span class="st">  </span><span class="er">|</span><span class="st">  </span><span class="dv">97</span>%</span>
<span id="cb16-46"><a href="#cb16-46"></a>  <span class="op">|</span><span class="st">                                                                            </span></span>
<span id="cb16-47"><a href="#cb16-47"></a><span class="st">  </span><span class="er">|======================================================================|</span><span class="st"> </span><span class="dv">100</span>%</span></code></pre></div>
<p>The result is stored in a file.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1"></a><span class="kw">head</span>(</span>
<span id="cb17-2"><a href="#cb17-2"></a>  <span class="kw">read.table</span>(<span class="kw">file.path</span>(<span class="kw">tempdir</span>(), <span class="st">&quot;ctcisQTL.out.txt&quot;</span>),</span>
<span id="cb17-3"><a href="#cb17-3"></a>             <span class="dt">header =</span> <span class="ot">TRUE</span>,</span>
<span id="cb17-4"><a href="#cb17-4"></a>             <span class="dt">sep =</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>))</span>
<span id="cb17-5"><a href="#cb17-5"></a><span class="co">#&gt;        term   response celltype      estimate  statistic    p.value</span></span>
<span id="cb17-6"><a href="#cb17-6"></a><span class="co">#&gt; 1 rs6678176 cg19251656     CD4T -0.0031975545 -0.3112062 0.75693773</span></span>
<span id="cb17-7"><a href="#cb17-7"></a><span class="co">#&gt; 2 rs6678176 cg19251656     CD8T  0.0029236713  0.4368269 0.66411762</span></span>
<span id="cb17-8"><a href="#cb17-8"></a><span class="co">#&gt; 3 rs6678176 cg19251656       NK  0.0021900746  0.6992125 0.48765929</span></span>
<span id="cb17-9"><a href="#cb17-9"></a><span class="co">#&gt; 4 rs6678176 cg01433766     CD4T -0.0007853244 -0.4642848 0.64445898</span></span>
<span id="cb17-10"><a href="#cb17-10"></a><span class="co">#&gt; 5 rs6678176 cg01433766     CD8T -0.0006336713 -0.5359356 0.59437935</span></span>
<span id="cb17-11"><a href="#cb17-11"></a><span class="co">#&gt; 6 rs6678176 cg01433766       NK -0.0012465110 -1.9903623 0.05203197</span></span></code></pre></div>
<p>The first 3 lines show the result for the association of SNP rs6678176 with CpG site cg19251656. Line 1 shows that the CD4T-specific effect of the SNP is -0.003. Since the p.value is 0.75, this is not significant.</p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
